trx/io.py

```diff
@@ -0,0 +1,74 @@
+#! /usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import os
+import logging
+
+from nibabel.streamlines.array_sequence import ArraySequence
+import numpy as np
+
+try:
+    import dipy
+    from dipy.io.stateful_tractogram import StatefulTractogram
+    from dipy.io.streamline import load_tractogram, save_tractogram
+    from dipy.io.utils import is_header_compatible
+    dipy_available = True
+except ImportError:
+    dipy_available = False
+
+from trx.utils import split_name_with_gz
+from trx.trx_file_memmap import load, save, TrxFile
+
+
+def load_sft_with_reference(filepath, reference=None,
+                            bbox_check=True):
+    if not dipy_available:
+        logging.error('Dipy library is missing, cannot use functions related '
+                      'to the StatefulTractogram.')
+        return None
+    # Force the usage of --reference for all file formats without an header
+    _, ext = os.path.splitext(filepath)
+    if ext == '.trk':
+        if reference is not None and reference != 'same':
+            logging.warning('Reference is discarded for this file format '
+                            '{}.'.format(filepath))
+        sft = load_tractogram(filepath, 'same',
+                              bbox_valid_check=bbox_check)
+    elif ext in ['.tck', '.fib', '.vtk', '.dpy']:
+        if reference is None or reference == 'same':
+            raise IOError('--reference is required for this file format '
+                          '{}.'.format(filepath))
+        else:
+            sft = load_tractogram(filepath, reference,
+                                  bbox_valid_check=bbox_check)
+
+    else:
+        raise IOError('{} is an unsupported file format'.format(filepath))
+
+    return sft
+
+
+def load_wrapper(tractogram_filename, reference):
```

Comment:
Should we rename this `load`? It is a wrapper around other things, but I think that it would be nice to be able to write the following code: 

```
from trx.io import load
tgm = load(file_path)
```

Reply:
But should the trx load function be renamed `load_trx`?

I would like to be able to differentiate both when reading the code?
I could switch to `import trx.trx_file_memmap as tmm` and use `tmm.load` too.

Which one do you prefer?